DECISION-RESEARCH.md

```diff
@@ -0,0 +1,315 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Trailblaze (September 2020)](#trailblaze-september-2020)
+- [Prototyping (July 2020)](#prototyping-july-2020)
+  - [/entity/item/Q42 vs /entity/Q42](#entityitemq42-vs-entityq42)
+  - [Consistent "default edit parameters"](#consistent-default-edit-parameters)
+  - [Pagination](#pagination)
+  - [Filtering rows](#filtering-rows)
+  - [Updating resources](#updating-resources)
+  - [Entities + Statements](#entities--statements)
+  - [RevisionIds (Path? Query?)](#revisionids-path-query)
+  - [Conditional Requests](#conditional-requests)
+  - [Resources in Plural vs Singular](#resources-in-plural-vs-singular)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Trailblaze (September 2020)
+
+## Title / Revision info in response bodies
+
+**Decision:Leave this meta infomation out of the response bodies for the first iteration.**
+
+- `lastrevid` & `modified`
+  - We already get an ETag with the revision ID, and a usable timestamp, that may cover the usage of this meta data
+- `title` & `pageid`
+  - Used to jump from Wikibase entities to titles for use with the MediaWiki core APIs (deletion & protection etc)
+    - Deletion is now provided in our REST API
+    - We could provide further APIs for other functionality such as protection, undeletion etc.
+  - Could be used for linking, but `Special:EntityPage` is also an option there
+- `ns`
+  - Will be easily available via the WikibaseManifest REST API in the future.
+  - The ID itself is probably not that useful anyway
+
+# Prototyping (July 2020)
+
+Research and decision record from the July 2020 prototyping week.
+
+## /entity/item/Q42 vs /entity/Q42
+
+**Decision: /entity/item/Q42 TBA why...**
```

Comment:
Documentation about this is very sparse but when considering this, word quickly came to (not implying that's a compelling train of thought) the topic of "search" and the fact that in the current (action API) one can not search for multiple entity types at the same time. This probably inspired, as we also mentioned that we can imagine "search" to happen on the entity collection, that the different entity types should form individual buckets, instead of a shared bucket.

I'm still skeptical about this because
* the entity type _name_ (e.g. "item") and the _prefix_ (e.g. "Q") are somewhat redundant
* the entity type _name_(s) could be surprisingly hard/expensive to come by in clients
* if search becomes a thing on the collections it essentially prevents ever having mixed results, while not having the different buckets would keep this option open

Reply:
We could probably bikeshed around this topic for some time, I'm starting to have no strong thoughts in either direction.
One thought is that if we want to change something like this in the future because of newly realized restrictions etc then we should be able to do that with ease. (Especially around the topic of search, as I think that might be one of the highly sought after features that ends up falling in scope (I could be wrong though)).

Now that we have restructured lots of the spec internals and also the way that he spec is displayed within swagger its starting to look like removing these extra entity type path elements might lead to an more readable spec thus easier to use API.

Reply:
> We could probably bikeshed around this topic for some time

Don't have to. Just tried to add my view given the 

> TBA why